lib/logstash/filters/geoip.rb

```diff
@@ -189,4 +181,15 @@ def get_geo_data_for_ip(ip)
       geo_data
     end
   end
+
+  def ensure_database!
+    # Use thread-local access to GeoIP. The Ruby GeoIP module forces a mutex
+    # around access to the database, which can be overcome with :pread.
+    # Unfortunately, :pread requires the io-extra gem, with C extensions that
+    # aren't supported on JRuby. If / when :pread becomes available, we can stop
+    # needing thread-local access.
+    if !Thread.current.key?(@threadkey)
+      Thread.current[@threadkey] = ::GeoIP.new(@database)
```

Comment:
@andrewvc Not sure how this would fix the issue since we just moved the code closer  to the lookup. Why Thread.current would be different than the `#filter` method?